File: backend/src/main/java/ru/bowling/bowlingapp/DTO/FreeMechanicApplicationRequestDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FreeMechanicApplicationRequestDTO {

    @NotBlank
    private String phone;

    @NotBlank
    @Size(min = 8, message = "Password must be at least 8 characters long")
    private String password;

    @NotBlank
    private String fullName;

    @NotNull
    private LocalDate birthDate;

    @NotNull
    private Integer educationLevelId;

    @NotBlank
    private String educationalInstitution;

    @NotNull
    @PositiveOrZero
    private Integer totalExperienceYears;

    @NotNull
    @PositiveOrZero
    private Integer bowlingExperienceYears;

    @NotNull
    private Boolean isEntrepreneur;

    @NotNull
    private Integer specializationId;

    @NotBlank
    private String region;

    @NotBlank
    private String skills;

    @NotBlank
    private String advantages;

    private Long clubId;

    private List<MechanicCertificationDTO> certifications;

    private List<MechanicWorkHistoryDTO> workHistory;
}

File: frontend/lib/models/free_mechanic_application_request_dto.dart
import 'mechanic_certification_dto.dart';
import 'mechanic_work_history_dto.dart';

class FreeMechanicApplicationRequestDto {
  final String phone;
  final String password;
  final String fullName;
  final DateTime birthDate;
  final int educationLevelId;
  final String educationalInstitution;
  final int totalExperienceYears;
  final int bowlingExperienceYears;
  final bool isEntrepreneur;
  final int specializationId;
  final String region;
  final String skills;
  final String advantages;
  final List<MechanicCertificationDto> certifications;
  final List<MechanicWorkHistoryDto> workHistory;
  final int? clubId;

  FreeMechanicApplicationRequestDto({
    required this.phone,
    required this.password,
    required this.fullName,
    required this.birthDate,
    required this.educationLevelId,
    required this.educationalInstitution,
    required this.totalExperienceYears,
    required this.bowlingExperienceYears,
    required this.isEntrepreneur,
    required this.specializationId,
    required this.region,
    required this.skills,
    required this.advantages,
    this.certifications = const [],
    this.workHistory = const [],
    this.clubId,
  });

  Map<String, dynamic> toJson() => {
        'phone': phone,
        'password': password,
        'fullName': fullName,
        'birthDate': birthDate.toIso8601String().split('T').first,
        'educationLevelId': educationLevelId,
        'educationalInstitution': educationalInstitution,
        'totalExperienceYears': totalExperienceYears,
        'bowlingExperienceYears': bowlingExperienceYears,
        'isEntrepreneur': isEntrepreneur,
        'specializationId': specializationId,
        'region': region,
        'skills': skills,
        'advantages': advantages,
        'certifications': certifications.map((e) => e.toJson()).toList(),
        'workHistory': workHistory.map((e) => e.toJson()).toList(),
        'clubId': clubId,
      };
}
